cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -227,6 +227,13 @@ impl Environment for ManagedEnvironment {
         self.ensure_locked(flox, &mut local_checkout)
     }
 
+    /// Returns the lockfile if it already exists.
+    fn existing_lockfile(&self) -> Result<Option<Lockfile>, EnvironmentError> {
+        CoreEnvironment::new(self.path.join(ENV_DIR_NAME), self.include_fetcher.clone())
+            .existing_lockfile()
+            .map_err(EnvironmentError::Core)
+    }
+
```

Comment:
suggestion: create `manifest.*` files from current generation/floxmetaif they dont exist yet.



```suggestion
    /// Returns the lockfile if it already exists.
    fn existing_lockfile(&self) -> Result<Option<Lockfile>, EnvironmentError> {
        let local_checkout = self.local_env_or_copy_current_generation(flox)?;
        local_checkout
            .existing_lockfile()
            .map_err(EnvironmentError::Core)
    }

```